web/controllers/v1/group_controller.ex

```diff
@@ -67,19 +67,21 @@ defmodule Cog.V1.GroupController do
   end
 
   def delete(conn, %{"id" => id}) do
-    with {:ok, group} <- Groups.by_id(id) do
-      case Groups.delete(group) do
-        {:ok, _} ->
-          send_resp(conn, :no_content, "")
-        {:error, :not_found} ->
-          conn
-          |> put_status(:not_found)
-          |> json(%{errors: "Group not found"})
-        {:error, :bad_id} ->
-          conn
-          |> put_status(:bad_request)
-          |> json(%{errors: "Bad ID format"})
-      end
+    results = with {:ok, group} <- Groups.by_id(id) do
+      Groups.delete(group)
+    end
```

Comment:
The repository takes a model and not an id. We could change that to accept an id instead, but I feel that there was a reason it was done this way. I just don't remember what it was.